autofetch/views.py

```diff
@@ -0,0 +1,137 @@
+from django.shortcuts import render
+from external.craigslist import fetch_craigslist_housing
+from external.googleapi.fetch import fetch_reverse_geocode
+from location.models import Location
+from location.models import Apartment
+from datetime import datetime
+from celery import shared_task
+from external.googleapi.g_utils import normalize_us_address
+import googlemaps
+
+
+CITY_LIST = ["brx", "brk", "fct", "lgi", "mnh", "jsy", "que", "stn", "wch"]
+
+
+def autofetch(request):
+
+    # The huge query: /autofetch/?city=brx,brk,fct,lgi,mnh,jsy,que,stn,wch&limit=1000000
+    if request.method == "GET":
+        # allow get param in url /autofetch/?city=brk,brx&limit=100
+
+        city_list, city, limit = set(), list(), 100
+
+        if "city" in request.GET:
+            city = request.GET.get("city").split(",")
+
+        if "limit" in request.GET:
+            limit = request.GET.get("limit")
+
+        for c in city:
+            if c in CITY_LIST:
+                city_list.add(c)
+
+        city_list = sorted(
+            city_list
+        )  # sorted for pass the test by having foreseeable order
+
+        bckgrndfetch.delay(list(city_list), int(limit))
+
+    return render(request, "account.html")
+
+
+@shared_task
+def bckgrndfetch(city_list, limit):
```

Comment:
how often is this configured to run? or do we have to manual start it?

Reply:
This annotation is for celery to mark the task it should assign to the worker to walk on. It will automatically start to listen to whoever calls this method right after our server is booted.

Reply:
ah I see, so calling the `/autofetch` url kicks off a background worker to update the locations?

Reply:
Right, note that call with the `city=` and `limit` parameter.